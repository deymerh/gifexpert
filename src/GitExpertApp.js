import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GitExpertApp = (defaultCategories = []) => {
  // const [categories, setCategories] = useState(['One punch']);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GitExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      { categories.map(categorie =>
        <GifGrid
          key={categorie}
          category={categorie}
        />)}
    </>
  )
}

export default GitExpertApp;