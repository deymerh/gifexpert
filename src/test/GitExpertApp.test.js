import '@testing-library/jest-dom';
import React from 'react';
import GitExpertApp from '../GitExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas en el compoente <GItExpertApp />', () => {
  const wrapper = shallow(<GitExpertApp />);
  test('Debe mostrar el conponente <GItExpertApp /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar una lista de categorias', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });

});