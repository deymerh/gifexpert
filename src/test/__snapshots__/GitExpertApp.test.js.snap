// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el compoente <GItExpertApp /> Debe mostrar el conponente <GItExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One punch"
    key="One punch"
  />
</Fragment>
`;
